### DISCUSSION TITLE
Should I give treatment to all people whose predict uplift is positive or just the portion which qini curve is ascending?

### DISCUSSION THREAD

**Discussion by Ramlinbird** [uber/causalml#655] [General]
Hello, I use the causalml's plot_qini function to draw the qini curve of my dataframe, the result is below, (The dataframe contains control/treatment random-ab-test data)
![image](https://github.com/uber/causalml/assets/11084132/68ee9984-ee79-422e-a8e6-beef0b79e865)
According to the curve, it seems that only the first 50% users with treatment is useful, and there is negative effect at the last.
But in the original predict uplift with treatment by the model, nearly 90% is larger than 0.
Should I chose the 90% or 30% to give the treatment? Thanks a lot.

**Comment by zeromh** [comment]
With a qini curve, you're checking the quality of your uplift model's rankings against *actual* uplift values. If your check with actual values shows that the last X% of your population has negative lift, then you probably don't want to target them even if your model predicts positive lift for those observations. Models aren't perfect, so this is a check.
